x-pack/test/functional/apps/dataset_quality/dataset_quality_privileges.ts

```diff
@@ -0,0 +1,237 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import expect from '@kbn/expect';
+import { DatasetQualityFtrProviderContext } from './config';
+import { getInitialTestLogs, getLogsForDataset } from './data';
+
+export default function ({ getService, getPageObjects }: DatasetQualityFtrProviderContext) {
+  const PageObjects = getPageObjects([
+    'common',
+    'security',
+    'navigationalSearch',
+    'observabilityLogsExplorer',
+    'datasetQuality',
+  ]);
+  const security = getService('security');
+  const synthtrace = getService('logSynthtraceEsClient');
+  const testSubjects = getService('testSubjects');
+  const find = getService('find');
+  const to = new Date(new Date().setDate(new Date().getDate() - 1)).toISOString();
+
+  const apacheAccessDatasetName = 'apache.access';
+  const apacheAccessDatasetHumanName = 'Apache access logs';
+
+  describe('Dataset quality handles user privileges', () => {
+    before(async () => {
+      await PageObjects.observabilityLogsExplorer.setupInitialIntegrations();
+    });
+
+    after(async () => {
+      await PageObjects.observabilityLogsExplorer.removeInstalledPackages();
+    });
+
+    describe('User cannot read any logs-*', () => {
+      before(async () => {
+        // Index logs for synth-* and apache.access datasets
+        await synthtrace.index(getInitialTestLogs({ to, count: 4 }));
+
+        await createDatasetQualityUserWithRole(security, 'dataset_quality_no_read', []);
+        await PageObjects.security.forceLogout();
```

Comment:
i believe this should be part of the after block

Reply:
The `await PageObjects.security.forceLogout();` here is to logout so that a new user can login. I've added logging out in `after` as well.